Form/Type/ColorSelectorType.php

```diff
@@ -33,7 +33,7 @@ public function setDefaultOptions(OptionsResolverInterface $resolver)
      */
     public function configureOptions(OptionsResolver $resolver)
     {
-        $resolver->setDefaults(array(
+        $defaults = array(
             'choices' => Colors::getAll(),
```

Comment:
Let me re-explain then.

# Use case 1 : the user just uses the bundle

1. They upgrade to the next minor version, that contains your PR
2. They see a deprecation notice, that comes from the SonataCoreExtension class
3. They ignore it, it works just like before
4. They prepare for the next upgrade
5. They set the parameter to `false`, the new codepath is used, and the deprecation notice disappears
6. They change their code to accomodate the fact that there are two options less now.
7. They do the major upgrade happily


# Use case 2 : the user directly calls Colors::getAll() from their code

1. They upgrade to the next minor version, that contains your PR
2. They see a deprecation notice, that comes from the Colors class
3. They ignore it, it works just like before
4. They prepare for the next upgrade
5. They call the method like this : Colors::getAll(false)
6. They change their code to accomodate the fact that there are two options less now.
7. They do the major upgrade happily

Right now you have just covered "Use case 2", which is kind of secondary. You still need to cover Use case 1, which is the most important.

Is any step unclear?